pkg/build/api/v1/types.go

```diff
@@ -52,8 +52,75 @@ type BuildSpec struct {
+	Info []*BuildTriggerInfo `json:"info,omitempty" description:"contains specific information about the triggers that fired a build"`
+}
+
+// BuildTriggerInfo contains information about why/who/what triggered a build.
+type BuildTriggerInfo struct {
+	// Details is a placeholder for  information about who triggered a build if
+	// it's either a user, a token or something else
+	Details BuildTriggerDetails `json:"details,omitempty" description:"generic information about a trigerred build and/or about the user that triggered it"`
+
+	// CIJobDetails is a generic placeholder that should store information about
+	// jobs from CI plugins that integrate with openshift and triggered a
+	// specific build
+	CIDetails CIJobDetails `json:"ciDetails,omitempty" description:"CI job information that triggered a build"`
+
+	// GitWebHookInfo holds information about a build that was triggered by a
+	// gitwebhook including its refs, branch, last commit id, etc. It also stores
+	// the type of plugin that triggered it, if it was a generic one, github,
+	// bitbucket, etc.
+	GitWebHook GitWebHookInfo `json:"gitWebHook,omitempty" description:"general information about a webhook that triggered this build"`
+
+	// ImageChangeBuildInfo - Stores info about the image that triggered a new
+	// build
+	ImageChange ImageChangeBuildInfo `json:"imageChange,omitempty" description:"information about an image change build trigger"`
+
+	// ConfigChangeBuildInfo - stores info about the last version of the config
+	// that triggered this build
+	ConfigChange ConfigChangeBuildInfo `json:"configChange,omitempty" description:"generic information about a configchange that triggered a build"`
+}
+
+// BuildTriggerDetails holds information about who/what triggered a build
+type BuildTriggerDetails string
+
+// CIJobDetails holds information about the CI job that triggered a
+// build
+type CIJobDetails struct {
+	// Type is type of the CI that started this. "jenkins", "travis", etc.
+	Type string `json:"type,omitempty" description:"type of CI that triggered the build"`
+
+	// JobURL holds the url that points to the job that triggered a specific
+	// build
+	JobURL string `json:"jobURL,omitempty"  description:"the URL with information about the specific job tha triggered a build"`
+}
+
+// GitWebHookInfo holds information about a git webhook that triggered a build
+type GitWebHookInfo struct {
+	// GitTriggerType is the git webhook trigger type (github, generic, etc)
+	GitTriggerType BuildTriggerType `json:"gitTriggerType,omitempty"`
+	// Git is where the information about the current git webhook is stored
+	GitDetails *GitInfo `json:"gitDetails,omitempty"`
+}
+
+// ImageChangeBuildInfo holds information about the image that triggered a new
+// build
+type ImageChangeBuildInfo struct {
+	// PreviousImageID is the old image ID before the image change trigger fires
+	PreviousImageID string `json:"previousImageID,omitempty" description:"the image ID before the image change trigger fired"`
```

Comment:
I'm pretty sure that I'm allowed to change the `From` of an `ImageChangeTrigger` on my BC which means that unless this a fully qualified pull spec with sha, this is insufficient to find the previous image.  Even it was, why would I choose to store that information differently instead of using an `ObjectReference`?